Source/Core/Core/CoreTiming.cpp

```diff
@@ -355,6 +357,16 @@ void CoreTimingManager::Throttle(const s64 target_cycle)
 
   m_throttle_last_cycle = target_cycle;
 
+#ifdef USE_RETRO_ACHIEVEMENTS
+  if (Config::Get(Config::RA_HARDCORE_ENABLED) &&
+      Config::Get(Config::MAIN_EMULATION_SPEED) < 1.0f &&
+      Config::Get(Config::MAIN_EMULATION_SPEED) > 0.0f)
+  {
+    Config::SetCurrent(Config::MAIN_EMULATION_SPEED, 1.0f);
```

Comment:
Does this update the UI?

I'm assuming you are setting this to avoid log spam.

I wonder why these config settings weren't set in `RefreshConfig`, seems like it'd be better to set a local variable there and modify the local variable instead of overwriting config.

Reply:
This does seem to immediately update the UI to return back to 100% yes. If the config settings should be moved, sure, but that seems a bit outside my scope for this particular PR.

Reply:
Will see how this goes:  https://github.com/dolphin-emu/dolphin/pull/12057